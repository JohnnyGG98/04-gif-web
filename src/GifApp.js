import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifApp = () => {

  // const categorias = ['One Punch', 'Samurai X', 'Naruto'];

  const [categorias, setCategoria] = useState([
    'One Punch'
  ]);

  // const handleAdd = () => {
  //   const newValue = 'Hunter X Hunter';
  //   // setCategoria([...categorias, newValue]);
  //   setCategoria(cats => [...cats, newValue]);
  // }

  return (
    <>
      <h2>Gif Expert App</h2>

      <AddCategory setCategoria={setCategoria}/>

      <hr/>

      <ol>
        { categorias.map( c => {
          return <GifGrid 
            key={c} 
            categoria={c} />
        }) }
      </ol>

    </>
  );
}

export default GifApp;