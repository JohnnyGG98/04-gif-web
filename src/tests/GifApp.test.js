import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import GifApp from '../GifApp';

describe('<GifApp /> | Main aplication', () => {
  
  test('Render', () => {
    const wrapper = shallow( <GifApp /> );

    expect( wrapper ).toMatchSnapshot();
  });

  test('Revisamos si se renderiza uno ', () => {
    const wrapper = shallow( <GifApp /> );

    expect( wrapper.find('GifGrid').length ).toBe(1);
  });
  

});
