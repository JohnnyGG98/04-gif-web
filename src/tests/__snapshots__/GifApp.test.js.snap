// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifApp /> | Main aplication Render 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategoria={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
